File: go/2575.go
package main

func divisibilityArray(word string, m int) []int {
	n := len(word)
	ans := make([]int, n)
	num := 0
	for i, ch := range word {
		num = (num*10 + int(ch)) % m
		if num == 0 {
			ans[i] = 1
		}
	}
	return ans
}

File: python/2575.py
from typing import List

class Solution:
    def divisibilityArray(self, word: str, m: int) -> List[int]:
        n = len(word)
        ans = [0] * n
        num = 0
        for i, ch in enumerate(word):
            num = (num * 10 + int(ch)) % m
            if num == 0:
                ans[i] = 1
        return ans

